diff --git i/gnuradio-runtime/CMakeLists.txt w/gnuradio-runtime/CMakeLists.txt
index 51eb3b0f8..3aaf5b45f 100644
--- i/gnuradio-runtime/CMakeLists.txt
+++ w/gnuradio-runtime/CMakeLists.txt
@@ -139,8 +139,8 @@ endif(ENABLE_GR_LOG AND HAVE_LOG4CPP)
 add_subdirectory(include)
 add_subdirectory(lib)
 add_subdirectory(apps)
+add_subdirectory(swig)
 if(ENABLE_PYTHON)
-     add_subdirectory(swig)
      add_subdirectory(python)
     add_subdirectory(examples)
 endif(ENABLE_PYTHON)
